src/shared/file_op.c

```diff
@@ -446,11 +446,15 @@ int CreatePID(const char *name, int pid)
     fprintf(fp, "%d\n", pid);
 
     if (chmod(file, 0640) != 0) {
+        merror(CHMOD_ERROR, file, errno, strerror(errno));
         fclose(fp);
         return (-1);
     }
 
-    fclose(fp);
+    if (fclose(fp)) {
+        merror("Could not write PID file '%s': %s (%d)", file, strerror(errno), errno);
```

Comment:
When there is no space in the device and we show this error, the PID files are deleted except the `wazuh-modulesd-<id>.pid`. After running `ossec-control start` we get the following error:
```
2018/06/22 12:47:47 wazuh-modulesd: ERROR: Could not write PID file '/var/ossec/var/run/wazuh-modulesd-17831.pid': No space left on device (28)
2018/06/22 12:47:47 wazuh-modulesd: CRITICAL: Couldn't create PID file: (No space left on device)
```
But the PID file is the only one that survives: 
```
root@ubuntu18:~# ll /var/ossec/var/run/
total 4
drwxrwx--- 2 root ossec   60 jun 22 12:48 ./
drwxr-x--- 6 root ossec 4096 jun 22 12:48 ../
-rw-r----- 1 root root     0 jun 22 12:47 wazuh-modulesd-17831.pid
root@ubuntu18:~#
```
We could force to remove the file if is not written by `unlink(file);`